src/memory-cache.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2018 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ not
+ * use this file except in compliance with the License. You may obtain a copy
+ of
+ * the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ under
+ * the License.
+ */
+
+'use strict';
+
+import * as Koa from 'koa';
+
+type CacheEntry = {
+  saved: Date,
+  expires: Date,
+  headers: string,
+  payload: string,
+};
+
+export const CACHE_DURATION_MINUTES = 60 * 24;
+
+export class MemoryCache {
+  private store: {[id: string]: CacheEntry} = {};
+
+  async clearCache() {
+    this.store = {};
+  }
+
+  cacheContent(key: string, headers: {}, payload: Buffer) {
```

Comment:
Suggest `headers: {[key: string]: string}` to make sure the values are only strings.